### ISSUE TITLE
Custom timing functions per property

### ISSUE THREAD

**Issue by jlongster** [lkzhao/Hero#94]
Is there a way to specify custom timing functions per property? For example, I want to use `easeInOut` to position but `easeOut` for fade. Seems pretty hard to do with the current string syntax, but have you thought about it at all?

**Comment by lkzhao** [comment]
We can definitely implement this. Let .duration & .timingFunction take in a modifier. Easy to do in code but hard to parse it from string though. Need to modify the parsing code to accommodate this.

Ideally the string syntax would support 

duration(0.5, scale(0.5))

I also wish to make some improvement to the string parsing code. To report error and provide a more swift enum-like syntax.

That's my idea at the moment.
